src/components/Timetable.jsx

```diff
@@ -31,190 +39,151 @@ const propTypes = {
   screenSize: PropTypes.number,
 }
 
-class Timetable extends React.Component {
+function calculateEvents (events, timeline) {
+  return R.map((event) => {
+    const {position, length} = calculateEventPosition(event, timeline)
+    return {
+      ...event,
+      _position: position,
+      _length: length,
+      _appear: eventAppearance(event),
+    }
+  }, events)
+}
 
-  /**
-   * Hides the days element by removing its animation property class
-   */
-  hide () {
-    const el = this.refs.days
+function createHourLabels (timeline, layout) {
+  console.log(timeline)
+  return calculateHourLabels(timeline).map((label) => {
+    console.log(label)
+    return <HourLabel key={label.id} position={label.position} length={label.length} layout={layout}>
+      {label.label}
+    </HourLabel>
+  })
+}
 
-    // Replay CSS animation
-    el.classList.remove('a-left')
-    el.classList.remove('a-right')
-  }
+function createDays (props, dayCount, events, eventsfn, animationDirection) {
+  const groupedEvents = groupEventsByDays(events)
 
-  /**
-   * Replays the CSS animation of all events from right side to the left.
-   */
-  animateLeft () {
-    const el = this.refs.days
+  return R.times((n) => {
+    return (
+      <Day id={n} key={'day-' + n} dayNum={dateInFuture(props.viewDate, n)} viewDate={props.viewDate}>
```

Comment:
I recommend overhauling <Day> props.

- Pass in the whole `Date` object, let the component decide how to display it.
- Instead of passing in the viewDate, just let the component know whether it is current (or active).
- `id` cannot be purely numerical; use the same value as in `key` prop.